### PR TITLE
Add "tunnel" mode to tsh proxy db

### PR DESCRIPTION
**Pull request by r0mant**
This PR updates `tsh proxy db` command to support "mutual TLS" mode where it opens a TLS tunnel which is already authenticated with the database's client certificate. This makes it possible for the actual database clients to connect through the local proxy directly without TLS authentication since the tunnel is already authenticated. The primary use-case for this is better UX for the local proxy usage with database clients which is esp. important for simplifying the Teleport Terminal usage.

Here's a usage example for Postgres. Start local authenticated proxy:

```
➜  ~ tsh proxy db --tunnel local
Started authenticated local proxy for the database "local" on 127.0.0.1:57129.

Use the following command to connect to the database:
  $ /opt/homebrew/opt/libpq/bin/psql postgres://localhost:57129
```

Connect to the database:

```
➜  ~ /opt/homebrew/opt/libpq/bin/psql postgres://localhost:57129/test
psql (14.2, server 13.5 (Debian 13.5-1.pgdg110+1))
Type "help" for help.

test=#
```

I have verified this for Postgres, MySQL, MongoDB and Redis. SQL Server already worked this way.

Refs https://github.com/gravitational/webapps.e/issues/177.

### CODE REVIEW

**Comment by ravicious** [comment]
To test this I need to have a cluster running code from this PR that also has a db connected to it, right? Is there an easy way to set this up?

**Review by smallinsky — APPROVED** [review]

File: tool/tsh/proxy.go [hunk 1]
```diff
@@ -351,3 +385,13 @@ Use following credentials to connect to the {{.database}} proxy:
   cert_file={{.cert}}
   key_file={{.key}}
 `))
+
+// dbProxyAuthTpl is the message that's printed for an authenticated db proxy.
+var dbProxyAuthTpl = template.Must(template.New("").Parse(
+	`Started authenticated local proxy for the database "{{.database}}" on {{.address}} with the following credentials:
+  cert_file={{.cert}}
+  key_file={{.key}}
```
[the remark this answers is not in the document]
Comment (r0mant) [reply tool/tsh/proxy.go:393]: Yeah good point, I also was mixed on whether to show them or not. I'll remove.

File: tool/tsh/proxy.go [hunk 2]
```diff
@@ -351,3 +383,11 @@ Use following credentials to connect to the {{.database}} proxy:
   cert_file={{.cert}}
   key_file={{.key}}
 `))
+
+// dbProxyAuthTpl is the message that's printed for an authenticated db proxy.
+var dbProxyAuthTpl = template.Must(template.New("").Parse(
+	`Started authenticated local proxy for the database "{{.database}}" on {{.address}}.
+
+Use the following command to connect to the database:
+  $ {{.command}}
```
Comment (greedy52) [inline tool/tsh/proxy.go:392]: The "mtls" mode should also work for GUIs? Not suggesting any change here. Just curious if this should be default mode eventually.

**Review by greedy52 — APPROVED** [review]
what happens when "mtls" is used against older servers?

**Comment by klizhentas** [comment]
@r0mant sorry for the nitpicks on the name, I was looking at this flag, what do you think about `--tunnel` mode? For users that could explain things a bit better.

File: tool/tsh/proxy.go (hunk 2 above)
Comment (r0mant) [reply tool/tsh/proxy.go:392]: Yep. I was planning to look into updating the docs after this merges.

**Comment by r0mant** [comment]
> what happens when "mtls" is used against older servers?

@greedy52 It will work for Mongo/Redis/MSSQL because those already use mutual TLS but not for Postgres/MySQL because older servers don't support plain connections through the TLS tunnel.

**Review by ravicious — APPROVED** [review]
I tested this on a Postgres db through Teleterm and it works like a charm.

In the future (after the Teleterm preview release) we might remove the concept of a "gateway" in Teleterm in favor of just running `tsh proxy db` underneath. Though the current gateway system also executes the equivalent of `tsh db login`, so we'll see.

**Pull request merged**